### ISSUE TITLE
Outdated stub in index

### ISSUE THREAD

**Issue by SendiMyrkr** [mesour-intellij-plugins/neon-nette-issues#22]
PHPStorm: 2021.3.2
Plugin: 1.0.2

I have no idea what is reason for this error or what caused it.

```
Outdated stub in index: file://C:/***/****.php indexing timestamp = 1643970472676, binary = false, byte size = 4196, char size = 4253, doc=DocumentImpl[file://C:/***/****.php], docSaved=true, wasIndexedAlready=true, queried at 1643970472676
doc length=4196
file length=4196
cached PSI class com.jetbrains.php.lang.psi.PhpFileImpl
PSI length=4196
projects with file: 1
physical file exists; length = 4196

java.lang.Exception
	at com.intellij.psi.stubs.StubTreeLoaderImpl.diagnoseLengthMismatch(StubTreeLoaderImpl.java:181)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.checkLengthMatch(StubTreeLoaderImpl.java:147)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readFromVFile(StubTreeLoaderImpl.java:114)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readOrBuild(StubTreeLoaderImpl.java:39)
	at com.intellij.psi.impl.source.PsiFileImpl.getStubTree(PsiFileImpl.java:634)
	at com.intellij.psi.impl.source.PsiFileImpl.getGreenStubTree(PsiFileImpl.java:956)
	at com.intellij.psi.impl.source.PsiFileImpl.getStubbedSpine(PsiFileImpl.java:255)
	at com.intellij.psi.stubs.StubProcessingHelperBase.lambda$getAllSpines$0(StubProcessingHelperBase.java:85)
	at com.intellij.util.containers.ContainerUtil.map(ContainerUtil.java:1851)
	at com.intellij.psi.stubs.StubProcessingHelperBase.getAllSpines(StubProcessingHelperBase.java:85)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:65)
	at com.intellij.psi.stubs.StubIndexImpl.lambda$processElements$2(StubIndexImpl.java:300)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:346)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:100)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:88)
	at com.jetbrains.php.PhpIndexImpl.getElements(PhpIndexImpl.java:592)
	at com.jetbrains.php.PhpIndexImpl.getUseAliasesByReferenceName(PhpIndexImpl.java:488)
	at com.jetbrains.php.lang.findUsages.PhpAliasReferenceSearcher.findUses(PhpAliasReferenceSearcher.java:95)
	at com.jetbrains.php.lang.findUsages.PhpAliasReferenceSearcher.processQuery(PhpAliasReferenceSearcher.java:62)
	at com.jetbrains.php.lang.findUsages.PhpAliasReferenceSearcher.processQuery(PhpAliasReferenceSearcher.java:31)
	at com.intellij.openapi.application.QueryExecutorBase.execute(QueryExecutorBase.java:77)
	at com.intellij.util.ExecutorsQuery.processResults(ExecutorsQuery.java:28)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:91)
	at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:108)
	at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:91)
	at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:108)
	at com.intellij.util.UniqueResultsQuery.processResults(UniqueResultsQuery.java:37)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:91)
	at com.intellij.util.AbstractQuery.forEach(AbstractQuery.java:83)
	at com.intellij.util.Query.anyMatch(Query.java:80)
	at com.mesour.intellij.neon.n.b.a(SourceFile:47)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:158)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:39)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:227)
	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:42)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:227)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:114)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:44)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:228)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:28)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:72)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:155)
	at com.mesour.intellij.neon.n.b.a(SourceFile:45)
	at com.mesour.intellij.neon.n.b.isImplicitUsage(SourceFile:26)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedSymbolsCache.isUsedImplicitlyOrFromCache(PhpUnusedSymbolsCache.java:47)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass$UnusedElementsData.searchForCandidateReferencingFiles(PhpUnusedGlobalSymbolsPass.java:265)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass.searchForUnusedSymbols(PhpUnusedGlobalSymbolsPass.java:161)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass.collectUnusedSymbols(PhpUnusedGlobalSymbolsPass.java:97)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass.doCollectInformation(PhpUnusedGlobalSymbolsPass.java:80)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:414)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1084)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:407)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:624)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:698)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:646)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:623)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:406)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:382)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:181)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:380)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:184)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by mesour** [comment]
Maybe I have to increase index version after every update. Next week will be next version for download.

**Comment by SendiMyrkr** [comment]
Just for information, i use this version since the release and it happend first to me today.

**Comment by mesour** [comment]
Fixed in 1.0.3
